src/osemgrep/targeting/Glob_matcher.mli

```diff
@@ -0,0 +1,90 @@
+(*
+   AST and matching of a glob pattern against a path.
+   This is purely syntaxic: the file system is not accessed.
+*)
+
+(* The location of a pattern, for logging and troubleshooting. *)
+type loc = {
+  (* File name or other source location name useful to a human reader
+     in error messages. *)
+  source_name : string;
+  (* Line number, starting from 1. *)
+  line_number : int;
+  line_contents : string;
+}
+
+(* Create a location from a pattern string rather than a location in a file. *)
+val string_loc : ?source_name:string -> string -> loc
+val show_loc : loc -> string
+
+type char_class_range = Class_char of char | Range of char * char
+[@@deriving show]
+
+type char_class = { complement : bool; ranges : char_class_range list }
+[@@deriving show]
+
+type component_fragment =
+  | Char of char
+  | Char_class of char_class
+  | Question
+  | Star
+[@@deriving show]
+
+(* A path component is what represents a simple file name in a directory *)
+type component =
+  | Component of component_fragment list
+  | Ellipsis (* '**' = path ellipsis *)
```

Comment:
I think I wanted an English name that distinguishes it from StarStar because we have a StarStar token that becomes Ellipsis if it's alone between two slashes or two successive stars if it's not alone. I understand that Ellipsis is confusing. Perhaps `Segment_repeat` would work.

Reply:
I'm calling it `Any_subpath` (until I change my mind again, of course).